src/main/kotlin/com/zopa/ktor/opentracing/OpenTracingClient.kt

```diff
@@ -10,14 +10,23 @@ import io.opentracing.propagation.Format
 import io.opentracing.tag.Tags
 
 
-class OpenTracingClient {
-    class Config
+class OpenTracingClient(
+        val tagsToExtractFromPath: Map<String, Regex>
+) {
+    class Configuration {
+        val tagsToExtractFromPath: MutableMap<String, Regex> = mutableMapOf(uuidTagAndReplace)
+
+        fun extractTagFromPath(tagName: String, regex: Regex) {
+            tagsToExtractFromPath[tagName] = regex
+        }
+    }
```

Comment:
I would prefer if we don't add config to the client feature. It's a better experience for users to have all config in one place. You can have an internal variable (similar to 'threadLocalSpanStack`) that you can write to in the server feature and read from here in the client feature.

Reply:
What if your application isn't a ktor server?

Reply:
Good point! The ScopeManager implementation requires the use of server feature unfortunately.

Reply:
Pushed dfc0640